examples/barotropic_tide.jl

```diff
@@ -0,0 +1,285 @@
+# # Barotropic tide
+#
+# In this example, we simulate the evolution of a barotropic tide over a hill.
+#
+# ## Install dependencies
+#
+# First let's make sure we have all required packages installed.
+
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, CairoMakie"
+# ```
+
+using Oceananigans
+using Oceananigans.Units
+
+# ## Grid
+
+# We use an immersed boundary two-dimensional grid (in ``x``--``z``) that is is periodic in
+# the ``x``-direction. To construct an immersed boundary grid we first need to create what
+# we refer to as "underlying grid", which the grid that encompasses the immersed boundary.
```

Comment:
"underlying" and "encompasses" in some way mean the opposite thing, so I think this explanation can be improved.

I would stick to describing how the code works. In the code, `ImmersedBoundaryGrid` is a wrapper around an underlying grid (not the other way around).

I would say something more concise like:

```
We create an `ImmersedBoundaryGrid` wrapped around an underlying two-dimensional `RectilinearGrid` that is periodic in `x` and bounded in `z`.
```